src/pages/SettingsPage.js

```diff
@@ -72,6 +72,22 @@ const Settings = ({ toRealmExplorer, currentUserPasswordHash, requestStorageWrit
     if (!ValidUrl.isWebUri(newSyncURL)) ToastAndroid.show('Not a valid URL', ToastAndroid.LONG);
     else setState({ ...state, modalKey: '', syncURL: newSyncURL });
   };
+
+  const onCheckConnection = () => {
+    //  Hash the password.
+    const AUTH_ENDPOINT = '/sync/v3/site';
+    const authURL = `${syncURL}${AUTH_ENDPOINT}`;
+    const syncSiteName = UIDatabase.getSetting(SETTINGS_KEYS.SYNC_SITE_NAME);
+    const syncAuthenticator = new SyncAuthenticator(AppSettings);
+    syncAuthenticator
+      .authenticate(authURL, syncSiteName, null, currentUserPasswordHash)
+      .then(() => {
+        ToastAndroid.show('Connection OK', ToastAndroid.LONG);
+      })
+      .catch(error => {
+        ToastAndroid.show(error.message, ToastAndroid.LONG);
+      });
```

Comment:
Maybe it would be best to check `error.message` against these messages: https://github.com/sussol/authentication-utilities/blob/master/src/index.js#L5-L9

That way, we can know more precisely what the problem is. Incorrect URL, password or if there is just no internet?

Reply:
the utility only says 'Unable to connect' in my experience, regardless of the actual error